tests/test_build.py

```
@@ -0,0 +1,69 @@
+        """Test that Streamlit server can start and respond to requests"""
+        # Correct path to main.py in the src directory
+        app_path = os.path.join(
+            os.path.dirname(os.path.dirname(os.path.abspath(__file__))),
+            "src",
+            "main.py",
+        )
+
+        project_root = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+
+        # Start Streamlit server in background
+        process = subprocess.Popen(
```

Ruff (0.12.2)

`subprocess` call: check for execution of untrusted input

(S603)